### PR TITLE
fix plotting with transposed nondim coords.

### LINKED ISSUE #3138: Plotting with 2d coordinates, error in transposed coordinate
This is similar to issue #1503, but specifically deals with requiring the correct order of dimensions for plotting 2d coordinates. An example of the issue is:

```
# create an example dataset
dates = pd.date_range('2000-01-01', '2001-12-31', name='time')
times = dates - dates[0]

x = np.linspace(0, 10, 101)

h = np.linspace(3, 7, 101)
s = np.linspace(0, 1, 51)
z = s[:, np.newaxis] * h[np.newaxis, :]

data = (np.sin(x) * np.cos(z)) * np.cos(np.asarray(times.days[:, np.newaxis, np.newaxis]))

# create an xarray dataset

ds = xr.Dataset({'data': (('time', 's', 'x'), data)}, 
                {'time':dates, 'x':x, 's':s, 'z':(('s', 'x'), z)})
ds.coords['zt'] = ds.z.transpose()
```
plotting with `z` and `zt` should give the same results, since the dimensions are clearly labeled and other than the order of the dimensions, they are exactly the same. 

This works:
```
ds.data[0].plot(x='x', y='z')
```

This does not:
```
ds.data[0].plot(x='x', y='zt')
```

Broadcasting, e.g., does not give a similar error. Specifically, this is true:
```
np.allclose((ds.data * ds.zt).values, (ds.data * ds.z).values)
```

### PR DESCRIPTION
**Pull request by dcherian**
- [x] Closes #3138
 - [x] Tests added
 - [x] Passes `black . && mypy . && flake8`
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

### CODE REVIEW

**Comment by max-sixty** [comment]
I don't know this code well, but LGTM

File: xarray/plot/plot.py [hunk 1]
```diff
@@ -672,10 +672,22 @@ def newplotfunc(
 
         # check if we need to broadcast one dimension
         if xval.ndim < yval.ndim:
-            xval = np.broadcast_to(xval, yval.shape)
+            dims = darray[ylab].dims
+            if xval.shape[0] == yval.shape[0]:
+                xval = np.broadcast_to(xval[:, np.newaxis], yval.shape)
```
Comment (dcherian) [inline xarray/plot/plot.py]: Is there a cleaner way of doing this?

**Comment by dcherian** [comment]
I'll merge tomorrow evening if nobody has any comments

**Comment by dcherian** [comment]
cc @hetland

**Comment by max-sixty** [comment]
Thanks @dcherian !

**Comment by hetland** [comment]
Looks like it was an easy fix. Thanks! I'll check it out soon.

**Pull request merged**